app/models/developers/external_profiles/linkedin_profile_fetcher.rb

```diff
@@ -1,40 +1,32 @@
 module Developers::ExternalProfiles
   class LinkedinProfileFetcher
     def developer_profiles
-      profiles = []
       developers = developers_with_linkedin_profiles
       Rails.logger.info "Fetching #{developers.count} #{"LinkedIn profile".pluralize(developers.count)}..."
 
       developers.find_each.with_index do |developer, index|
         response = fetch_linkedin_profile(developer.linkedin)
         record = external_profile(developer, response)
-        profiles << record if record.present?
+        Developers::ExternalProfile.upsert(record, unique_by: [:developer_id, :site]) if record.present?
 
         developers_left = developers.count - index - 1
         Rails.logger.info "#{developers_left} #{"profile".pluralize(developers_left)} left"
       end
-
-      upsert_external_profiles(profiles)
     end
 
     def external_profile(developer, response)
       if response.error?
         {developer_id: developer.id, site: "linkedin", data: {}, error: response.error}
       else
         external_profile = Developers::ExternalProfile.linkedin_developer(developer)
-        external_profile.data = response.data unless external_profile.blank?
+        response_data = response.data.delete_if { |key, value| key == "logo_url" } unless response.data.blank?
```

Comment:
Removing logo_url from response as it is adding signature in value which is different in each fetch and creating issue when we compare data to update it or not.

Reply:
Good call!